src/Compilers/Core/CodeAnalysisTest/DefaultAnalyzerAssemblyLoaderTests.cs

```diff
@@ -28,642 +37,846 @@ public class AssemblyLoadTestFixtureCollection : ICollectionFixture<AssemblyLoad
-            {
-                var ex = Assert.ThrowsAny<Exception>(() => analyzer.GetType().GetMethod("Method")!.Invoke(analyzer, new object[] { sb }));
-                Assert.True(ex is MissingMethodException or TargetInvocationException, $@"Unexpected exception type: ""{ex.GetType()}""");
-            }
-            else
-            {
-                analyzer.GetType().GetMethod("Method")!.Invoke(analyzer, new object[] { sb });
-                Assert.Equal("42", sb.ToString());
-            }
+                if (ExecutionConditionUtil.IsCoreClr)
+                {
+                    var ex = Assert.ThrowsAny<Exception>(() => analyzer.GetType().GetMethod("Method")!.Invoke(analyzer, new object[] { sb }));
+                    Assert.True(ex is MissingMethodException or TargetInvocationException, $@"Unexpected exception type: ""{ex.GetType()}""");
+                }
+                else
+                {
+                    analyzer.GetType().GetMethod("Method")!.Invoke(analyzer, new object[] { sb });
+                    Assert.Equal("42", sb.ToString());
+                }
+            });
         }
 
         [Fact]
         public void AssemblyLoading_AnalyzerReferencesSystemCollectionsImmutable_02()
         {
-            StringBuilder sb = new StringBuilder();
+            Run(static (DefaultAnalyzerAssemblyLoader loader, AssemblyLoadTestFixture testFixture) =>
+            {
+                StringBuilder sb = new StringBuilder();
 
-            var loader = new DefaultAnalyzerAssemblyLoader();
-            loader.AddDependencyLocation(_testFixture.UserSystemCollectionsImmutable.Path);
-            loader.AddDependencyLocation(_testFixture.AnalyzerReferencesSystemCollectionsImmutable2.Path);
+                loader.AddDependencyLocation(testFixture.UserSystemCollectionsImmutable.Path);
+                loader.AddDependencyLocation(testFixture.AnalyzerReferencesSystemCollectionsImmutable2.Path);
 
-            Assembly analyzerAssembly = loader.LoadFromPath(_testFixture.AnalyzerReferencesSystemCollectionsImmutable2.Path);
-            var analyzer = analyzerAssembly.CreateInstance("Analyzer")!;
-            analyzer.GetType().GetMethod("Method")!.Invoke(analyzer, new object[] { sb });
-            Assert.Equal(ExecutionConditionUtil.IsCoreClr ? "1" : "42", sb.ToString());
+                Assembly analyzerAssembly = loader.LoadFromPath(testFixture.AnalyzerReferencesSystemCollectionsImmutable2.Path);
+                var analyzer = analyzerAssembly.CreateInstance("Analyzer")!;
+                analyzer.GetType().GetMethod("Method")!.Invoke(analyzer, new object[] { sb });
+                Assert.Equal(ExecutionConditionUtil.IsCoreClr ? "1" : "42", sb.ToString());
+            });
         }
 
-        [ConditionalFact(typeof(WindowsOnly), typeof(CoreClrOnly))]
-        public void AssemblyLoading_NativeDependency()
+        [Fact(Skip = "https://github.com/dotnet/roslyn/issues/66104")]
```

Comment:
Why skip?

Reply:
Issue #66104 is what caused me to start digging into this and finding the issue in our test harness. Originally I had the fix for #66104 and the test fix in the same PR. After some thought decided to break them into diff PRs to make review easier. This is just disabled until the next PR.